packages/curve-ui-kit/src/features/connect-wallet/ui/ConnectWalletIndicator.tsx

```diff
@@ -11,9 +14,28 @@ export const ConnectWalletIndicator = ({
   onConnectWallet,
   onDisconnectWallet,
   ...props
-}: ConnectWalletIndicatorProps) =>
-  walletAddress ? (
-    <ConnectedWalletLabel walletAddress={walletAddress} onDisconnectWallet={onDisconnectWallet} {...props} />
-  ) : (
-    <ConnectWalletButton label={label} onConnectWallet={onConnectWallet} {...props} />
+}: ConnectWalletIndicatorProps) => {
+  const [isOpen, open, close] = useSwitch()
```

Comment:
we cannot `useSwitch` because then the other connect buttons in the app won't work. That's why we need to keep the modal state in the global state

Reply:
Right, forgot there's multiple ones. I tend to just have one and re-use it 🙈